daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -42,6 +44,11 @@ case object VariableUndefined extends VariableState
 
 case object VariableDefined extends VariableState
 
+/**
+ * Used to detect circular definitions of variables
+ */
```

Comment:
Need a more thurough description of this. Though, I'm wondering if we can get away with VariableInProcess and don't need a new state?